services/src/main/java/org/keycloak/protocol/oidc/utils/OIDCRedirectUriBuilder.java

```diff
@@ -47,8 +51,19 @@ protected OIDCRedirectUriBuilder(KeycloakUriBuilder uriBuilder) {
 
     public abstract Response build();
 
+    public static OIDCRedirectUriBuilder fromUri(String baseUri, OIDCResponseMode responseMode, KeycloakSession session,
+            ClientModel client) {
+        return fromUri(baseUri, responseMode, session, client, null);
+    }
+
+    public static OIDCRedirectUriBuilder fromUri(String baseUri, OIDCResponseMode responseMode, KeycloakSession session,
+            AuthenticatedClientSessionModel clientSession) {
+        return fromUri(baseUri, responseMode, session, clientSession.getClient(), clientSession);
+    }
 
-    public static OIDCRedirectUriBuilder fromUri(String baseUri, OIDCResponseMode responseMode, KeycloakSession session, AuthenticatedClientSessionModel clientSession) {
+    private static OIDCRedirectUriBuilder fromUri(String baseUri, OIDCResponseMode responseMode, KeycloakSession session,
```

Comment:
In theory, the client does not need to be passed as an argument to this method as it can be retrieved from KeycloakSession via the call to `session.getContext().getClient()` . However I am not 100% sure if the client is always set in the KeycloakContext. So leaving to you if you prefer to keep client as an argument or not considering that we are in time pressure...

Reply:
Sounds better. Changed.